sympy/core/tests/test_symbol.py

```diff
@@ -30,10 +30,18 @@ def test_Symbol():
 
 def test_Dummy():
     assert Dummy() != Dummy()
-    Dummy._count = 0
-    d1 = Dummy()
-    Dummy._count = 0
-    assert d1 == Dummy()
+
+
+def test_Dummy_force_dummy_index():
+    raises(AssertionError, lambda: Dummy(dummy_index=1))
+    assert Dummy('d', dummy_index=2) == Dummy('d', dummy_index=2)
+    assert Dummy('d1', dummy_index=2) != Dummy('d2', dummy_index=2)
+    d1 = Dummy('d', dummy_index=3)
+    d2 = Dummy('d')
+    # not only unlikely, in fact impossible (b/c we keep a list of dummy_index)
+    assert d1 != d2
```

Comment:
oops, this comment isn't strictly true any more with this version...  (test will still pass because the autogenerated dummy_index is `>= 10**6`.)